### ISSUE TITLE
main.ts issue after update

### ISSUE THREAD

**Issue by bubbleheadinc** [nstudio/xplat#74]
Just updated to xplat 7.2.3.

Running: 
`cd apps/nativescript-maintmobile && tns run ios --emulator --bundle --hmr`

Getting TSC errors in main.ts due to `<%` syntax and references to `routing` and `sample` not existing: 
```ERROR in app/main.ts(16,4): error TS1110: Type expected.
app/main.ts(16,6): error TS1109: Expression expected.
app/main.ts(16,31): error TS1109: Expression expected.
app/main.ts(16,32): error TS1109: Expression expected.
app/main.ts(19,4): error TS1110: Type expected.
app/main.ts(19,6): error TS1109: Expression expected.
app/main.ts(19,8): error TS1109: Expression expected.
app/main.ts(19,9): error TS1109: Expression expected.
app/main.ts(20,3): error TS1109: Expression expected.
```

Line 16 issue in main.ts:
```<% if (routing || sample) { %>
  // attach to livesync hooks and perform navigation
  require('@gp-mm/nativescript').attachLivesyncNavigation();
  <% } %>
  const hmrUpdate = require('nativescript-dev-webpack/hmr').hmrUpdate;
``` 

Full main.ts
```// https://github.com/NativeScript/nativescript-dev-webpack/issues/660#issuecomment-422711983
import 'reflect-metadata';
import { platformNativeScriptDynamic } from 'nativescript-angular/platform';
import { AppOptions } from 'nativescript-angular/platform-common';
import { enableProdMode } from '@angular/core';
import { AppModule } from './app.module';

// If built with env.uglify
declare const __UGLIFIED__;
if (typeof __UGLIFIED__ !== 'undefined' && __UGLIFIED__) {
  require('@angular/core').enableProdMode();
}

let options: AppOptions = {};
if (module['hot']) {
  <% if (routing || sample) { %>
  // attach to livesync hooks and perform navigation
  require('@gp-mm/nativescript').attachLivesyncNavigation();
  <% } %>
  const hmrUpdate = require('nativescript-dev-webpack/hmr').hmrUpdate;

  options.hmrOptions = {
    moduleTypeFactory: () => AppModule,
    livesyncCallback: platformReboot => {
      console.log('HMR: Sync...');
      hmrUpdate();
      setTimeout(platformReboot, 0);
    }
  };
  hmrUpdate();

  module['hot'].accept(['./app.module']);
}

platformNativeScriptDynamic(options).bootstrapModule(AppModule);
```

Package.json Deps:
```"dependencies": {
    "@angular/animations": "~7.1.0",
    "@angular/common": "~7.1.0",
    "@angular/compiler": "~7.1.0",
    "@angular/core": "~7.1.0",
    "@angular/forms": "~7.1.0",
    "@angular/http": "~7.1.0",
    "@angular/platform-browser": "~7.1.0",
    "@angular/platform-browser-dynamic": "~7.1.0",
    "@angular/router": "~7.1.0",
    "@ngrx/effects": "6.0.1",
    "@ngrx/router-store": "6.0.1",
    "@ngrx/store": "^7.1.0",
    "@ngx-translate/core": "~11.0.0",
    "@ngx-translate/http-loader": "~4.0.0",
    "@nrwl/nx": "7.5.0",
    "@types/underscore": "^1.8.9",
    "core-js": "^2.5.4",
    "moment": "^2.22.2",
    "nativescript-angular": "~7.1.0",
    "nativescript-ngx-fonticon": "^4.2.0",
    "nativescript-theme-core": "^1.0.4",
    "reflect-metadata": "^0.1.12",
    "rxjs": "^6.3.3",
    "rxjs-compat": "^6.2.2",
    "tns-core-modules": "~5.1.0",
    "underscore": "^1.9.1",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.6.1",
    "@angular/cli": "^7.2.2",
    "@angular/compiler-cli": "^6.1.0",
    "@angular/language-service": "^6.1.0",
    "@ngrx/store-devtools": "6.0.1",
    "@nrwl/schematics": "7.5.0",
    "@nstudio/schematics": "^7.2.3",
    "@types/jasmine": "2.8.6",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "codelyzer": "~4.2.1",
    "jasmine-core": "~2.99.1",
    "jasmine-marbles": "0.4.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~1.4.2",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "nativescript-dev-typescript": "^0.7.9",
    "nativescript-dev-webpack": "^0.19.0",
    "ngrx-store-freeze": "0.2.4",
    "prettier": "1.13.7",
    "protractor": "~5.3.0",
    "tns-platform-declarations": "~5.1.0",
    "ts-node": "~5.0.1",
    "tslint": "~5.9.1",
    "typescript": "~3.1.1"
  }```

**Comment by NathanWalker** [comment]
Thank you @bubbleheadinc did you run `ng update` by chance to auto update your `main.ts`?

**Comment by NathanWalker** [comment]
@bubbleheadinc When you ran `ng update`, the `7-0-1` migration ran which copied the latest main.ts template file with mustache. I just published `7.2.4` which adjusts that migration for those using `ng update` to ensure it clears the mustache template syntax from latest changes for future `ng update` users. Thanks for heads up.

On your end you can just open the `main.ts` and adjust this:
```
<% if (routing || sample) { %>
  // attach to livesync hooks and perform navigation
  require('@gp-mm/nativescript').attachLivesyncNavigation();
  <% } %>
```
to this:
```

  // attach to livesync hooks and perform navigation
  require('@gp-mm/nativescript').attachLivesyncNavigation();

```

Keep in mind that if your app does not have routing setup you can also just comment out the attachLivesyncNavigation call there. 👍

**Comment by bubbleheadinc** [comment]
@NathanWalker Thanks! That helped. Yes, I did `ng update` following: https://github.com/nstudio/xplat/wiki/Use-ng-update

Although I'm now getting a webpack error looking for attachLivesyncNavigation fnc (app does run with it commented out):

```
JavaScript error:
file:///app/bundle.js:5761:89: JS ERROR TypeError: __webpack_require__("../../../xplat/nativescript/index.ts").attachLivesyncNavigation is not a function. (In '__webpack_require__("../../../xplat/nativescript/index.ts").attachLivesyncNavigation()', '__webpack_require__("../../../xplat/nativescript/index.ts").attachLivesyncNavigation' is undefined)
(CoreFoundation) *** Terminating app due to uncaught exception 'NativeScript encountered a fatal error: TypeError: __webpack_require__("../../../xplat/nativescript/index.ts").attachLivesyncNavigation is not a function. (In '__webpack_require__("../../../xplat/nativescript/index.ts").attachLivesyncNavigation()', '__webpack_require__("../../../xplat/nativescript/index.ts").attachLivesyncNavigation' is undefined)
at
1   ./main.ts@file:///app/bundle.js:5761:89
2   __webpack_require__@file:///app/bundle.js:752:34
3   checkDeferredModules@file:///app/bundle.js:45:42
4   @file:///app/bundle.js:845:38
5   anonymous@file:///app/bundle.js:846:12
...
```
